File: typescript/70/70.climbing-stairs.ts
/*
 * @lc app=leetcode id=70 lang=typescript
 *
 * [70] Climbing Stairs
 */

// @lc code=start
function climbStairs(n: number): number {
    if (n == 1) {
        return 1
    }

    if (n == 2) {
        return 2
    }

    let step = [0, 1];
    let tmp = 0;

    for (let i = 0; i < n; i++) {
        tmp = step[1];
        step[1] += step[0];
        step[0] = tmp;
    }

    return step[1];
};
// @lc code=end

File: rust/src/bin/70/70.climbing-stairs.rs
/*
 * @lc app=leetcode id=70 lang=rust
 *
 * [70] Climbing Stairs
 */

// @lc code=start
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let mut arr = vec![0; n as usize + 2];
        arr[1] = 1;
        arr[2] = 2;

        if n <= 2 {
            return n
        }

        for i in 3..=(n as usize)+1 {
            arr[i] = arr[i - 1] + arr[i - 2]; 
        }

        return arr[n as usize]
    }
}
// @lc code=end
